src/routes/conversation/[id]/+server.ts

```diff
@@ -133,6 +142,9 @@ export async function POST({ request, locals, params, getClientAddress }) {
 		})
 		.parse(json);
 
+	if (usageLimits?.messageLength && (newPrompt?.length ?? 0) > usageLimits.messageLength) {
+		throw error(400, "Message too long.");
+	}
```

Comment:
We should probably prevent in the front-end a too-long message

Also make sure to convert \r\n to \n (if it's a native form submission) cc @krampstudio

Reply:
rate limits make sense to keep secret, but message length... :thinking: